hydrolib/core/io/polyfile/models.py

```diff
@@ -55,6 +55,31 @@ def _get_identifier(self, data: dict) -> Optional[str]:
         return f"x:{x} y:{y} z:{z}"
 
 
+class XYZValues:
+    """Class that holds lists of x, y and z values."""
+
+    x: List[float] = []
+    """The x values."""
+
+    y: List[float] = []
+    """The y values."""
+
+    z: List[Optional[float]] = []
+    """The z values. The values can be `None`."""
+
+    def append(self, x: float, y: float, z: Optional[float]):
```

Comment:
z: Optional[float]=None

Reply:
`Optional` in this case means that it can be either a float or a `None`. Below the surface `Optional[float]` actually is a `Union[float, None]`. So it does not mean that this argument is optional for the user; they have to provide it, but it is allowed to be `None`
For more information: https://docs.python.org/3/library/typing.html#typing.Optional

However, I have removed this function (see comment above).